MHVLib/Clients/MHVPlatformClient.m

```diff
@@ -115,39 +133,87 @@ - (void)newApplicationCreationInfoWithCompletion:(void(^)(MHVApplicationCreation
-                                        completion:(void(^_Nullable)(NSError *_Nullable error))completion
+- (void)getHealthRecordThingTypeDefinitionsWithTypeIds:(NSArray<NSString *> *_Nullable)typeIds
+                                              sections:(MHVThingTypeSections)sections
+                                            imageTypes:(NSArray<NSString *> *_Nullable)imageTypes
+                                 lastClientRefreshDate:(NSDate *_Nullable)lastClientRefreshDate
+                                            completion:(void(^)(NSDictionary<NSString *, MHVThingTypeDefinition *> *_Nullable definitions, NSError *_Nullable error))completion;
 {
-    MHVMethod *method = [MHVMethod removeApplicationRecordAuthorization];
-    method.recordId = recordId;
+    MHVASSERT_PARAMETER(completion);
+    
+    if (!completion)
+    {
+        return;
+    }
+    
+    MHVThingTypeDefinitionRequestParameters *parameters = [[MHVThingTypeDefinitionRequestParameters alloc]initWithTypeIds:typeIds
+                                                                                                                 sections:sections
+                                                                                                               imageTypes:imageTypes
+                                                                                                    lastClientRefreshDate:lastClientRefreshDate];
+    
+    MHVMethod *method = [MHVMethod getThingType];
+    method.parameters = [XSerializer serializeToString:parameters withRoot:@"info"];
     
     [self.connection executeHttpServiceOperation:method
                                       completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error)
     {
-        if (completion)
+        if (error)
         {
-            completion(error);
+            completion(nil, error);
+            
+            return;
         }
+        
+        NSMutableDictionary<NSString *, MHVThingTypeDefinition *> *definitions = [NSMutableDictionary new];
+        
+        XReader *reader = [[XReader alloc] initFromString:response.infoXml];
+        
+        if ([reader readStartElementWithName:@"info"])
+        {
+            NSMutableArray<MHVThingTypeDefinition *> *typeDefinitions = [reader readElementArray:@"thing-type" asClass:[MHVThingTypeDefinition class]];
+            
+            for (NSString *typeId in typeIds)
+            {
+                for (MHVThingTypeDefinition *definition in typeDefinitions)
+                {
+                    if ([typeId caseInsensitiveCompare:definition.typeId.UUIDString] == NSOrderedSame)
+                    {
+                        [definitions setObject:definition forKey:typeId];
+                        
+                        [typeDefinitions removeObject:definition];
+                        
+                        break;
+                    }
+                }
+            }
```

Comment:
Unfortunately, the casing of the UUID string is not guaranteed to be uppercase or lowercase, so I need to do a case insensitive comparison.